draft-yorgos-dnsop-dry-run-dnssec.md

```diff
@@ -0,0 +1,320 @@
+capitals, as shown here.
+
+dry-run DS
+: The DS record with the special DS type digest algorithm that signals dry-run
+  DNSSEC for the delegation.
+
+real DS
+: The actual DS record for the delegation. Replaces the dry-run DS to complete
+  DNSSEC deployment.
+
+dry-run zone
+: A zone that is DNSSEC signed but uses a dry-run DS to signal the use of the
+  dry-run DNSSEC method.
+
+wet-run client
+: A client that has opted-in to receive the actual DNSSEC errors from the
+  upstream validating resolver instead of the insecure answers.
+
+
+# Description {#description}
+
+TODO
+
+## The dry-run DS structure {#dry-run-structure}
+
+The dry-run DS record is a normal DS record with updated semantics to allow for
+dry-run signaling to a validating resolver. The DS Type Digest Algorithm value
+MUST be TBD (DRY-RUN). The first octet of the DS Digest field contains the
+actual Type Digest Algorithm, followed by the actual Digest:
+
+
+                        1 1 1 1 1 1 1 1 1 1 2 2 2 2 2 2 2 2 2 2 3 3
+    0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |           Key Tag             |  Algorithm    |    DRY-RUN    |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   | Digest Type   |                                               /
+   +-+-+-+-+-+-+-+-+            Digest                             /
+   /                                                               /
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+
+Validating resolvers encountering such a DS record will know to mark this
+delegation as dry-run DNSSEC and extract the actual Type Digest Algorithm and
+Digest from the dry-run DS Digest field.
+
+Validating resolvers that have no knowledge for the DRY-RUN DS Type Digest
+Algorithm MUST disregard the DS record as per [@RFC6840, see, section 5.2].
+
+## DNSSEC Error Reporting {#dnssec-error-reporting}
+
+The main purpose of the dry-run DNSSEC proposal is to be able to monitor
+potential DNS breakage when adopting DNSSEC for a zone. The main tool to do
+that is DNS Error Reporting [@DNS-ERROR-REPORTING].
+
+Operators that want to use dry-run DNSSEC SHOULD support DNSSEC Error Reporting
+and have a reporting agent in place to receive the error reports.
+
+Implementations that support dry-run DNSSEC MUST also support DNSSEC Error
+Reporting and report any DNSSEC errors for the dry-run zone to the
+designated report agent.
+
+## Parent zone records {#parent-zone-records}
+
+The only change that needs to happen for dry-run DNSSEC is for the parent to
+be able to publish the dry-run DS record. If the parent accepts DS records from
+the child, the child needs to provide the dry-run DS record. If the parent does
+not accept DS records and generates the DS records from the DNSKEY, support for
+generating the dry-run DS record when needed should be added to the parent if
```

Comment:
```suggestion
generating the dry-run DS record, when needed, should be added to the parent if
````
This makes this sentence easier to read imo

Reply:
Added